lib/devicetrust/native/device_windows.go

```diff
@@ -78,106 +79,69 @@ func handleTPMActivateCredential(encryptedCredential, encryptedCredentialSecret
-	)
-	// ThinkPad P P14s:
-	// PS > Get-WmiObject Win32_BIOS | Select -ExpandProperty SerialNumber
-	// PF47WND6
-	out, err := cmd.Output()
-	if err != nil {
+	type Win32_BIOS struct {
+		SerialNumber string
+	}
+
+	var bios Win32_BIOS
+	query := wmi.CreateQuery(&bios, "")
+	if err := wmi.Query(query, &bios); err != nil {
 		return "", trace.Wrap(err)
 	}
-	return string(bytes.TrimSpace(out)), nil
+
+	return bios.SerialNumber, nil
 }
 
 func getReportedAssetTag() (string, error) {
-	cmd := exec.Command(
-		"powershell",
-		"-NoProfile",
-		"Get-WmiObject Win32_SystemEnclosure | Select -ExpandProperty SMBIOSAssetTag",
-	)
-	// ThinkPad P P14s:
-	// PS > Get-WmiObject Win32_SystemEnclosure | Select -ExpandProperty SMBIOSAssetTag
-	// winaia_1337
-	out, err := cmd.Output()
-	if err != nil {
+	type Win32_SystemEnclosure struct {
+		SMBIOSAssetTag string
+	}
+
+	var system Win32_SystemEnclosure
+	query := wmi.CreateQuery(&system, "")
+	if err := wmi.Query(query, &system); err != nil {
 		return "", trace.Wrap(err)
 	}
-	return string(bytes.TrimSpace(out)), nil
+
+	return system.SMBIOSAssetTag, nil
 }
 
 func getDeviceModel() (string, error) {
-	cmd := exec.Command(
-		"powershell",
-		"-NoProfile",
-		"Get-WmiObject Win32_ComputerSystem | Select -ExpandProperty Model",
-	)
-	// ThinkPad P P14s:
-	// PS> Get-WmiObject Win32_ComputerSystem | Select -ExpandProperty Model
-	// 21J50013US
-	out, err := cmd.Output()
-	if err != nil {
+	type Win32_ComputerSystem struct {
+		Model string
+	}
+	var cs Win32_ComputerSystem
+	query := wmi.CreateQuery(&cs, "")
+	if err := wmi.Query(query, &cs); err != nil {
 		return "", trace.Wrap(err)
 	}
-	return string(bytes.TrimSpace(out)), nil
+
+	return cs.Model, nil
 }
 
 func getDeviceBaseBoardSerial() (string, error) {
-	cmd := exec.Command(
-		"powershell",
-		"-NoProfile",
-		"Get-WmiObject Win32_BaseBoard | Select -ExpandProperty SerialNumber",
-	)
-	// ThinkPad P P14s:
-	// PS> Get-WmiObject Win32_BaseBoard | Select -ExpandProperty SerialNumber
-	// L1HF2CM03ZT
-	out, err := cmd.Output()
-	if err != nil {
+	type Win32_BaseBoard struct {
+		SerialNumber string
+	}
+	var bb Win32_BaseBoard
+	query := wmi.CreateQuery(&bb, "")
+	if err := wmi.Query(query, &bb); err != nil {
 		return "", trace.Wrap(err)
 	}
 
-	return string(bytes.TrimSpace(out)), nil
+	return bb.SerialNumber, nil
 }
 
 func getOSVersion() (string, error) {
-	cmd := exec.Command(
-		"powershell",
-		"-NoProfile",
-		"Get-WmiObject Win32_OperatingSystem | Select -ExpandProperty Version",
-	)
-	// ThinkPad P P14s:
-	// PS>  Get-WmiObject Win32_OperatingSystem | Select -ExpandProperty Version
-	// 10.0.22621
-	out, err := cmd.Output()
-	if err != nil {
-		return "", trace.Wrap(err)
-	}
-
-	return string(bytes.TrimSpace(out)), nil
+	ver := windows.RtlGetVersion()
+	return fmt.Sprintf("%v.%v.%v", ver.MajorVersion, ver.MinorVersion, ver.BuildNumber), nil
 }
 
 func getOSBuildNumber() (string, error) {
```

Comment:
Maybe combine getOSVersion and getOSBuildNumber in a single method?

We might as well use WMI too, wdyt?

Reply:
I was thinking only use WMI when necessary, it's just more complicated..